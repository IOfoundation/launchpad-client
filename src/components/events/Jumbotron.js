import React from 'react';

const Jumbotron = () => {
  return (
    <div className="events-jumbotron">
      <div className="events-jumbotron__container">
        <h2 className="events-jumbotron__title">{"What's Happening"}</h2>
        <p className="events-jumbotron__description">
          {
            'There are many events throughout the Sacramento Region for all phases of businesses. See our most popular keynote events on the event list below or explore the calendar.'
          }
        </p>
      </div>
    </div>
  );
};

export default Jumbotron;
