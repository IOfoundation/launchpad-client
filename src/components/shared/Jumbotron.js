import React from 'react';
import {PropTypes} from 'prop-types';

const Jumbotron = props => {
  const {descriptionModifacor, title, description, background} = props;
  const descriptionClasses = ['jumbotron__description', 'text-thin'];

  if (props.descriptionModifacor) {
    descriptionClasses.push(`jumbotron__description--${descriptionModifacor}`);
  }

  return (
    <div
      className="jumbotron"
      style={{
        backgroundImage: `url(/static-data/images/${background}.png)`,
      }}
    >
      <div className="jumbotron__container">
        <h2 className="jumbotron__title">{title}</h2>
        <p className={descriptionClasses.join(' ')}>{description}</p>
      </div>
    </div>
  );
};

Jumbotron.propTypes = {
  background: PropTypes.string,
  description: PropTypes.string,
  descriptionModifacor: PropTypes.string,
  title: PropTypes.string,
};

export default Jumbotron;
