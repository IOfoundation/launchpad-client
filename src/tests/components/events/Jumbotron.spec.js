import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Jumbotron from '../../../components/events/Jumbotron';

describe('<Jumbotron />', () => {
  it('renders snapshot of Events Jumbotron', () => {
    const wrapper = shallow(<Jumbotron />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
