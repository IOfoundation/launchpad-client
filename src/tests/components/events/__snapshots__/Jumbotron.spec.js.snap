// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders snapshot of Events Jumbotron 1`] = `
<div
  className="events-jumbotron"
>
  <div
    className="events-jumbotron__container"
  >
    <h2
      className="events-jumbotron__title"
    >
      What's Happening
    </h2>
    <p
      className="events-jumbotron__description text-thin"
    >
      There are many events throughout the Sacramento Region for all phases of businesses. See our most popular keynote events on the event list below or explore the calendar.
    </p>
  </div>
</div>
`;
